server/sonar-server/src/main/java/org/sonar/server/component/index/ComponentIndex.java

```diff
@@ -107,6 +109,10 @@ private static TopHitsBuilder createSubAggregation(ComponentIndexQuery query) {
   private QueryBuilder createQuery(ComponentIndexQuery query, ComponentTextSearchFeature... features) {
     BoolQueryBuilder esQuery = boolQuery();
     esQuery.filter(authorizationTypeSupport.createQueryFilter());
+    Set<String> componentUuids = query.getComponentUuids();
+    if (!componentUuids.isEmpty()) {
+      esQuery.must(termsQuery("_id", componentUuids));
```

Comment:
`esQuery.must` will let the termsQuery participate in the scoring mechanism and cannot be cached by elasticsearch. Change it to `esQuery.filter`.

https://www.elastic.co/guide/en/elasticsearch/guide/current/_queries_and_filters.html#_performance_differences